File: ruby/resistor-color/resistor_color.rb
# frozen_string_literal: true

module ResistorColor
  COLORS = %w[
    black
    brown
    red
    orange
    yellow
    green
    blue
    violet
    grey
    white
  ].freeze

  module_function

  def color_code(color)
    COLORS.index(color)
  end
end

File: python/resistor-color/resistor_color.py
from typing import List


__COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white"
]


def color_code(color: str):
    return __COLORS.index(color)


def colors() -> List[str]:
    return __COLORS
